torch/jit/annotations.py

```diff
@@ -282,6 +300,8 @@ def try_ann_to_type(ann, loc):
         return DeviceObjType.get()
     if ann is torch.dtype:
         return IntType.get()  # dtype not yet bound in as its own type
+    if inspect.isclass(ann) and issubclass(ann, enum.Enum):
+        return EnumType(_qualified_name(ann), get_enum_value_type(ann, loc))
```

Comment:
Return None here to disable

Reply:
Added a check here, but raised an exception instead of returning None so that user gets a clearer error.